### PR TITLE
feat: allow templates in baseURL for openapi

### PR DESCRIPTION
**Pull request by jensneuse**

### CODE REVIEW

File: packages/sdk/src/v2openapi/index.ts [hunk 1]
```diff
@@ -674,10 +700,14 @@ class RESTApiBuilder {
...
 		}
 		const secure = this.spec.servers.find((server) => server.url.startsWith('https'));
-		if (secure) {
-			return secure.url;
-		}
-		return this.spec.servers[0].url;
+		return secure ? secure.url : this.spec.servers[0].url;
+	};
+	private cleanupBaseURL = (url: string): string => {
+		return url.replace(/{[a-zA-Z]+}/, (str) => {
```
Comment (StarpTech) [inline packages/sdk/src/v2openapi/index.ts:706]: This will fail `{ tld }` (spaces)

File: packages/sdk/src/v2openapi/index.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ import {
 	applyNameSpaceToTypeFields,
 } from '../definition/namespacing';
 import { mapInputVariable } from '../configure';
+import { resolveConfigurationVariable } from '../../dist/configure/variables';
```
Comment (StarpTech) [inline packages/sdk/src/v2openapi/index.ts:51]: You cant reference to dist :smile:

File: packages/sdk/src/v2openapi/index.ts [hunk 3]
```diff
@@ -226,6 +245,8 @@ class RESTApiBuilder {
 			ChildNodes: [],
 			Directives: [],
 		});
+		const baseURL = resolveConfigurationVariable(this.dataSources[this.dataSources.length - 1].Custom.Fetch.baseUrl!);
```
Comment (StarpTech) [inline packages/sdk/src/v2openapi/index.ts:248]: We should throw an error when the user passes `new EnvironmentVariables` in combination with base URL variables.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: packages/sdk/src/v2openapi/index.ts (hunk 1 above)
Comment (jensneuse) [reply packages/sdk/src/v2openapi/index.ts:706]: that's ok, it's invalid. we can add validation later

**Review by StarpTech — CHANGES_REQUESTED** [review]
Could you also update the docs?

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**